kotsadm/web/src/components/apps/AppDetailPage.jsx

```diff
@@ -139,14 +151,12 @@ class AppDetailPage extends Component {
       });
       if (res.ok && res.status == 200) {
         const app = await res.json();
-        this.setState({ app, loadingApp: false });
+        this.setState({ app, loadingApp: false, gettingAppErrMsg: "" });
       } else {
-        console.log("failed to get app, unexpected status code", res.status);
-        this.setState({ loadingApp: false });
+        this.setState({ loadingApp: false, gettingAppErrMsg: `failed to get app, unexpected status code: ${res.status}` });
```

Comment:
👍  all other functions should follow the same scenario as this one:

1- reset the error at the beginning and when the query succeeds
2- use response status code instead of `error`